processor/groupbytraceprocessor/README.md

```diff
@@ -38,6 +39,9 @@ The `num_traces` property tells the processor what's the maximum number of trace
 
 The `wait_duration` property tells the processor for how long it should keep traces in the internal storage. Once a trace is kept for this duration, it's then released to the next consumer and removed from the internal storage. Spans from a trace that has been released will be kept for the entire duration again.
 
+The `num_workers` property controls how many concurrent workers the processor will use to process traces. This number
+should be set equal to the amount of physical processors available to the collector.
```

Comment:
"set equal to" or "no greater than"?  Are there other considerations?  What if multiple `groupbytrace` processors are configured, as in the example?  Should the sum of the `num_workers` options be no greater than the number of processors or can each use up to that amount independently?  Why "physical" processors?  What about virtual environments that have only vCPUs?  What about hyperthreading?  Does "physical processor" mean the item deposited in a socket or does it mean processor core, of which there may be many in a single socket?

Reply:
Good questions. Could @pkositsyn or @jpkrohling can assist here with these questions around recommended configs. The default is 1 and is listed in `config.go` docs.

Reply:
My recollection is that @pkositsyn used this in production at the time and found this to be the optimal value for his case. The truth is that this is a regular go routine, which is not precisely tied to physical processors. Without further evidence, it's hard to state that this should always be lower, equal, or higher than the number of processors. Users should be encouraged to play with this number and find the best value for their scenario. For instance, if the next consumer of traces is a remote database, then it would certainly be beneficial to have N*num_procs, as networking I/O would undoubtedly be the bottleneck. If the next consumer is CPU-bound, the lower the number of routines we use here the less pressure we'll have on the overall process.

The default value reflects the state of this component before the number of workers has been added.

Reply:
For comparison, the [exporterhelper](https://github.com/open-telemetry/opentelemetry-collector/tree/v0.69.0/exporter/exporterhelper) has a similar property `sending_queue.num_consumers` and the docs don't describe any recommendations for its value. Perhaps it's good enough?

As a side note, I believe the docs should mention the default value of the property, something like "The default value is 1" or "(default = 1)" or similar.

Reply:
@jpkrohling answered very accurately - feel free to tune the parameter for performance, docs just give a decent baseline.

> Why "physical" processors?

Actually, you're right, it would be better to use GOMAXPROCS, which approximates the limit for linear growth in performance for CPU-bound operations (and this value should already be tuned in any environment). The default value 1 is indeed just to keep the compatible behavior with previous code. One may even argue, that `runtime.GOMAXPROCS(-1)` should be used as a default instead

> I believe the docs should mention the default value of the property

I agree, feel free to make pull request

> docs don't describe any recommendations for its value. Perhaps it's good enough?

Sending queue is all about I/O-bound operations, so I actually don't think that there really exists reasonable
advice - too many different parameters could affect the choice of workers number